### PR TITLE
Feature/bai 311 add nav buttons to docs

### PR DESCRIPTION
**Pull request by ARADDCC002**

### CODE REVIEW

File: src/docs/DocsWrapper.tsx [hunk 1]
```diff
@@ -22,14 +27,30 @@ const paddingIncrement = 2
...
+  const router = useRouter()
   const { docsMenuContent, errorMessage } = useContext(DocsMenuContext)
+  const [flattenedPages, setFlattenedPages] = useState<DocsMenuContent>([])
 
   const linkColour = useMemo(
     () => (theme.palette.mode === 'light' ? theme.palette.primary.main : theme.palette.secondary.main),
     [theme]
   )
 
+  const flattenPages = useCallback((array) => {
```
Comment (ARADDCC012) [inline src/docs/DocsWrapper.tsx:39]: Could update this to TS:
```
  const flattenPages = useCallback((array: DocsMenuContent) => {
    let result: DocsMenuContent = []
    array.forEach((item) => {
      result.push(item)
      if (isDocHeading(item)) {
        result = result.concat(flattenPages(item.children))
      }
    })
    return result.filter((item) => !isDocHeading(item) || item.hasIndex)
  }, [])
```

File: src/docs/DocsWrapper.tsx [hunk 2]
```diff
@@ -22,14 +27,30 @@ const paddingIncrement = 2
...
+    array.forEach((a) => {
+      result.push(a)
+      if (Array.isArray(a.children)) {
+        result = result.concat(flattenPages(a.children))
+      }
+    })
+    return result.filter((item: any) => item.hasIndex || item.hasIndex === undefined)
+  }, [])
+
+  useEffect(() => {
```
Comment (ARADDCC012) [inline src/docs/DocsWrapper.tsx:50]: You could condense this effect hook and the above state hook into a single memo hook:
```
  const flattenedPages = useMemo(() => flattenPages(docsMenuContent), [docsMenuContent, flattenPages])
```

File: src/docs/DocsWrapper.tsx [hunk 3]
```diff
@@ -67,20 +88,32 @@ export default function DocsWrapper({ children }: DocsWrapperProps): ReactElemen
...
-    [pathname]
+    [router.pathname]
   )
 
   const docsMenu = useMemo(
     () => docsMenuContent.map((doc) => createDocElement(doc)),
     [docsMenuContent, createDocElement]
   )
 
+  const reducedPath = router.pathname.replace(/^(\/docs\/)/, '')
```
Comment (ARADDCC012) [inline src/docs/DocsWrapper.tsx:105]: Could wrap both of these in a memo hook:
```
  const reducedPath = useMemo(() => router.pathname.replace(/^(\/docs\/)/, ''), [router.pathname])

  const currentIndex = useMemo(
    () => flattenedPages.findIndex((item) => item.slug === reducedPath),
    [flattenedPages, reducedPath]
  )
```

File: src/docs/DocsWrapper.tsx [hunk 4]
```diff
@@ -116,7 +149,31 @@ export default function DocsWrapper({ children }: DocsWrapperProps): ReactElemen
...
+                            Home
+                          </Button>
+                        )}
+                        {currentIndex > 0 && (
+                          <Button startIcon={<ArrowBack />} onClick={() => changePage(currentIndex - 1)}>
+                            {flattenedPages[currentIndex - 1].title}
+                          </Button>
+                        )}
+                        {currentIndex < flattenedPages.length - 1 && (
+                          <Button endIcon={<ArrowForward />} onClick={() => changePage(currentIndex + 1)} sx={{}}>
```
Comment (ARADDCC012) [inline src/docs/DocsWrapper.tsx:168]: Empty `sx` prop being provided here

File: src/docs/DocsWrapper.tsx [hunk 5]
```diff
@@ -22,14 +27,30 @@ const paddingIncrement = 2
 
 export default function DocsWrapper({ children }: DocsWrapperProps): ReactElement {
   const theme = useTheme()
-  const { pathname } = useRouter()
+  const router = useRouter()
```
Comment (ARADDCC012) [inline src/docs/DocsWrapper.tsx:30]: Is there a reason for reverting from destructuring? I tend to prefer destructuring as it reduces the amount of dot notation being used throughout the code.

**Review by ARADDCC012 — CHANGES_REQUESTED** [review]

File: src/docs/DocsWrapper.tsx (hunk 5 above)
Comment (ARADDCC002) [reply src/docs/DocsWrapper.tsx:30]: We need to use router itself to change page, L#114 for example.

**Review by ARADDCC012 — APPROVED** [review]

**Pull request merged**